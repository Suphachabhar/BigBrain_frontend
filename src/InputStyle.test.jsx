import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import { InputStyle } from './components/InputStyle.styles';

describe('InputStyle', () => {
  const noop = () => {};

  // test the inputStyle component
  it('trigger onChange event handler when clicked', () => {
    const onChange = jest.fn();
    shallow(<InputStyle onChange="{onChange}" />).simulate('change');
    expect(onChange).toHaveBeenCalledTimes(0);
  });

  it('renders with minimal props', () => {
    const selectFunc = renderer.create(<InputStyle onChange={noop} />).toJSON;
    expect(selectFunc).toMatchSnapshot();
  });
  it('renders with provided id (snapshot)', () => {
    const quizname = renderer.create(<InputStyle name={'title'} />).toJSON;
    expect(quizname).toMatchSnapshot();
  });
});
