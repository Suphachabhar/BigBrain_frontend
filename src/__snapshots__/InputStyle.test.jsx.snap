// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputStyle renders with minimal props 1`] = `[Function]`;

exports[`InputStyle renders with provided id (snapshot) 1`] = `[Function]`;
